src/lib/provable/gadgets/elliptic-curve.ts

```diff
@@ -129,7 +128,7 @@ function double(p1: Point, Curve: { modulus: bigint; a: bigint }) {
     let [x1_, y1_] = Field3.toBigints(x1, y1);
     let denom = inverse(mod(2n * y1_, f), f) ?? 0n;
 
-    let m = mod(3n * mod(x1_ ** 2n, f) * denom, f);
+    let m = mod((3n * mod(x1_ ** 2n, f) + Curve.a) * denom, f);
```

Comment:
In [PR310](https://github.com/zksecurity/o1js-bindings/blob/e081466d5e435fe3fbd32fa83ef5c0ef3f030dec/crypto/elliptic-curve.ts#L578) this `m` is computed with a single modular operation, meaning: `let m = mod((3n * x * x + a) * d, p);`. Cannot we remove one `mod` from this line?

Reply:
good question! I think using `mod()` twice is better actually, because both x and d are large field elements. so x^2 is already a bigint of 2x the length, and it's more efficient to first reduce it (with `mod()`) back to 1x the length before doing another bigint multiplication with d. In the code you mention, we get an intermediate result of 2x * 1x = 3x the length